index.bs

```diff
@@ -1550,16 +1541,26 @@ invoked, must run these steps:
 
 1. Let |environment| be the [=current settings object=].
 
-1. If |environment|'s [=environment settings object/origin=] is an [=opaque origin=],
+1. Let |map| be the result of running [=obtain a local storage bottle map=]
+   with |environment| and `"fileSystem"`. If this returns failure,
    return [=a promise rejected with=] a {{SecurityError}}.
 
-1. Let |storage| be |environment|'s [=environment settings object/origin=]'s [=storage unit=].
-1. Let |bucket| be |storage|'s [=bucket=].
+1. If |map|["root"] does not [=map/exist=]:
+  1. Set |map|["root"] to a new [=directory entry=].
```

Comment:
I think this would be slightly more readable with a temp var, e.g. 

1. Let _dir_ be a new directory entry.
1. Set _dir_'s foo to bar...
1. Set _dir_'s foo to bar...
1. Set |map|["root"] to _dir_

Reply:
Done